fuel-tx/src/transaction/consensus_parameters.rs

```diff
@@ -178,27 +239,107 @@ impl Default for ConsensusParameters {
 #[deprecated(since = "0.12.2", note = "use `ConsensusParameters` instead.")]
 pub mod default_parameters {
-    use super::ConsensusParameters;
+    use crate::{
+        transaction::consensus_parameters::{
+            PredicateParameters,
+            ScriptParameters,
+            TxParameters,
+        },
+        ContractParameters,
+        FeeParameters,
+    };
     use fuel_types::ChainId;
 
-    pub const CONTRACT_MAX_SIZE: u64 = ConsensusParameters::DEFAULT.contract_max_size;
-    pub const MAX_INPUTS: u64 = ConsensusParameters::DEFAULT.max_inputs;
-    pub const MAX_OUTPUTS: u64 = ConsensusParameters::DEFAULT.max_outputs;
-    pub const MAX_WITNESSES: u64 = ConsensusParameters::DEFAULT.max_witnesses;
-    pub const MAX_GAS_PER_TX: u64 = ConsensusParameters::DEFAULT.max_gas_per_tx;
-    pub const MAX_SCRIPT_LENGTH: u64 = ConsensusParameters::DEFAULT.max_script_length;
+    pub const CONTRACT_MAX_SIZE: u64 = ContractParameters::DEFAULT.contract_max_size;
+    pub const MAX_INPUTS: u64 = TxParameters::DEFAULT.max_inputs;
+    pub const MAX_OUTPUTS: u64 = TxParameters::DEFAULT.max_outputs;
+    pub const MAX_WITNESSES: u64 = TxParameters::DEFAULT.max_witnesses;
+    pub const MAX_GAS_PER_TX: u64 = TxParameters::DEFAULT.max_gas_per_tx;
+
+    pub const MAX_SCRIPT_LENGTH: u64 = ScriptParameters::DEFAULT.max_script_length;
     pub const MAX_SCRIPT_DATA_LENGTH: u64 =
-        ConsensusParameters::DEFAULT.max_script_data_length;
-    pub const MAX_STORAGE_SLOTS: u64 = ConsensusParameters::DEFAULT.max_storage_slots;
+        ScriptParameters::DEFAULT.max_script_data_length;
+
+    pub const MAX_STORAGE_SLOTS: u64 = ContractParameters::DEFAULT.max_storage_slots;
+
     pub const MAX_PREDICATE_LENGTH: u64 =
-        ConsensusParameters::DEFAULT.max_predicate_length;
+        PredicateParameters::DEFAULT.max_predicate_length;
     pub const MAX_PREDICATE_DATA_LENGTH: u64 =
-        ConsensusParameters::DEFAULT.max_predicate_data_length;
-    pub const MAX_GAS_PER_PREDICATE: u64 =
-        ConsensusParameters::DEFAULT.max_gas_per_predicate;
-    pub const GAS_PRICE_FACTOR: u64 = ConsensusParameters::DEFAULT.gas_price_factor;
-    pub const GAS_PER_BYTE: u64 = ConsensusParameters::DEFAULT.gas_per_byte;
+        PredicateParameters::DEFAULT.max_predicate_data_length;
     pub const MAX_MESSAGE_DATA_LENGTH: u64 =
-        ConsensusParameters::DEFAULT.max_message_data_length;
-    pub const CHAIN_ID: ChainId = ConsensusParameters::DEFAULT.chain_id;
+        PredicateParameters::DEFAULT.max_message_data_length;
+
+    pub const MAX_GAS_PER_PREDICATE: u64 =
+        PredicateParameters::DEFAULT.max_gas_per_predicate;
+    pub const GAS_PRICE_FACTOR: u64 = FeeParameters::DEFAULT.gas_price_factor;
+    pub const GAS_PER_BYTE: u64 = FeeParameters::DEFAULT.gas_per_byte;
+
+    pub const CHAIN_ID: ChainId = ChainId::new(0);
+}
+
+/// A collection of parameters for convenience
+#[derive(Debug, Clone, Copy)]
+pub struct ConsensusParams<'a> {
```

Comment:
It does make it harder to work with.

I had reasons I preferred references, but I'm not really convinced by them anymore, so I'll just get rid of the refs. Especially since all of them are `Copy`.